src/tests/test_openapi.py

```diff
@@ -13,12 +13,22 @@ def test_root_view(api_client, afval_dataset, fietspaaltjes_dataset, filled_rout
     response = api_client.get(url)
     assert response.status_code == 200, response.data
 
-    assert response.data == {
-        "datasets": {
-            "afvalwegingen": "http://testserver/v1/afvalwegingen/",
-            "fietspaaltjes": "http://testserver/v1/fietspaaltjes/",
-        }
-    }
+    # Prove that response containes datasets
+    assert "datasets" in response.data
+
+    # Prove that datasets contains the datasets and noting else
+    assert set(response.data["datasets"].keys()) == set(["afvalwegingen","fietspaaltjes"])  
+
+    # Prove that both dataset elements are dictionaries
+    assert type(response.data["datasets"]["afvalwegingen"]) == dict
+    assert type(response.data["datasets"]["fietspaaltjes"]) == dict
+
+    # Prove that both dataset elements have the required keys
+    keys_afvalwegingen = list(response.data["datasets"]["afvalwegingen"].keys()) 
+    keys_fietspaaltjes = list(response.data["datasets"]["fietspaaltjes"].keys()) 
+    required_keys = ["id", "name", "api_url"]
+    assert set(keys_afvalwegingen + required_keys) == set(keys_afvalwegingen)
```

Comment:
What does this prove actually? That `id`, `name`, and `api_url` are in both keys?
Then assert should be just `assert "id" in keys_fietspaaltjes` etc.

BUT: at the end it does not really prove that data is present and links are correct etc. 
Current test would be much more readable if it was just:

```
response = api_client.get(url)
assert response.status_code == 200, response.data

assert response.data == {
    "datasets": 
       "afvalwegingen": {
            "id": "afvalwegingen",
            ....
       },
       "fietspaaltjes": {...}
}
```

This would explicitly check if datasets are there, that keys are there and that data is present.

Reply:
Im trying to implement your suggestion. But the json contains absolute urls ("api_url": "http://localhost:8008/v1/fietspaaltjes/") Is there a way to get the absolute path without a request object?

Reply:
Inside the tests, the host in the url should be "testserver". See test_views.py.